usb/esp_tinyusb/tusb_msc_storage.c

```diff
@@ -0,0 +1,563 @@
+        ESP_ERROR_CHECK(tinyusb_msc_storage_mount(s_storage_handle->base_path));
+    }
+    return true;
+}
+
+// Invoked when received SCSI READ10 command
+// - Address = lba * BLOCK_SIZE + offset
+// - Application fill the buffer (up to bufsize) with address contents and return number of read byte.
+int32_t tud_msc_read10_cb(uint8_t lun, uint32_t lba, uint32_t offset, void *buffer, uint32_t bufsize)
+{
+    esp_err_t err = msc_storage_read_sector(lba, offset, bufsize, buffer);
+    if (err != ESP_OK) {
+        ESP_LOGE(TAG, "msc_storage_read_sector failed: 0x%x", err);
+        return 0;
+    }
+    return bufsize;
+}
+
+// Invoked when received SCSI WRITE10 command
+// - Address = lba * BLOCK_SIZE + offset
+// - Application write data from buffer to address contents (up to bufsize) and return number of written byte.
+int32_t tud_msc_write10_cb(uint8_t lun, uint32_t lba, uint32_t offset, uint8_t *buffer, uint32_t bufsize)
+{
+    esp_err_t err = msc_storage_write_sector(lba, offset, bufsize, buffer);
+    if (err != ESP_OK) {
+        ESP_LOGE(TAG, "msc_storage_write_sector failed: 0x%x", err);
+        return 0;
+    }
+    return bufsize;
+}
+
+/**
+ * Invoked when received an SCSI command not in built-in list below.
+ * - READ_CAPACITY10, READ_FORMAT_CAPACITY, INQUIRY, TEST_UNIT_READY, START_STOP_UNIT, MODE_SENSE6, REQUEST_SENSE
+ * - READ10 and WRITE10 has their own callbacks
+ *
+ * \param[in]   lun         Logical unit number
+ * \param[in]   scsi_cmd    SCSI command contents which application must examine to response accordingly
+ * \param[out]  buffer      Buffer for SCSI Data Stage.
+ *                            - For INPUT: application must fill this with response.
+ *                            - For OUTPUT it holds the Data from host
+ * \param[in]   bufsize     Buffer's length.
+ *
+ * \return      Actual bytes processed, can be zero for no-data command.
+ * \retval      negative    Indicate error e.g unsupported command, tinyusb will \b STALL the corresponding
+ *                          endpoint and return failed status in command status wrapper phase.
+ */
+int32_t tud_msc_scsi_cb(uint8_t lun, uint8_t const scsi_cmd[16], void *buffer, uint16_t bufsize)
+{
+    int32_t ret;
+
+    switch (scsi_cmd[0]) {
+    case SCSI_CMD_PREVENT_ALLOW_MEDIUM_REMOVAL:
+        /* SCSI_CMD_PREVENT_ALLOW_MEDIUM_REMOVAL is the Prevent/Allow Medium Removal
+        command (1Eh) that requests the library to enable or disable user access to
+        the storage media/partition. */
+        ret = 0;
+        break;
+    default:
+        ESP_LOGW(TAG, "tud_msc_scsi_cb() invoked: %d", scsi_cmd[0]);
+        tud_msc_set_sense(lun, SCSI_SENSE_ILLEGAL_REQUEST, SCSI_CODE_ASC_INVALID_COMMAND_OPERATION_CODE, SCSI_CODE_ASCQ);
+        ret = -1;
+        break;
+    }
+    return ret;
+}
+
+// Invoked when device is unmounted
+void tud_umount_cb(void)
+{
+    ESP_ERROR_CHECK(tinyusb_msc_storage_mount(s_storage_handle->base_path));
```

Comment:
We usually try to avoid aborting on an error inside IDF components. Is there no way to gracefully handle the error here, other than aborting?

Reply:
I have removed ESP_ERROR_CHECK from this file for now.

I can see that there are many assert statement being used in whole `idf-extra-components/usb`. I will discuss and plan to remove these in subsequent PR.